src/libopenrave/robot.cpp

```diff
@@ -77,7 +89,10 @@ void RobotBase::GripperInfo::DeserializeJSON(const rapidjson::Value& value, dRea
         if (memberName == "id" || memberName == "name" || memberName == "grippertype" || memberName == "gripperJointNames") {
             continue;
         }
-        orjson::SetJsonValueByKey(docGripperInfo, memberName, it->value);
+
+        // update objects recursively 
+        _PartialUpdate(docGripperInfo, memberName.c_str(), it->value, docGripperInfo.GetAllocator());
```

Comment:
@LLipter We already have `OpenRAVE::orjson::UpdateJson` why don't you use it instead?

Reply:
Or, add the recursive version of it to `openravejson.h` instead

Reply:
I moved the function to `openravejson.h` as you suggested. `OpenRAVE::orjson::UpdateJson` is not enough as it only updates the top-level key-value pair. We need to update all levels recursively.